### ISSUE TITLE
Logstash exception in parsing MAC address

### ISSUE THREAD

**Issue by salva-ferrer** [logstash-plugins/logstash-codec-netflow#26]
Hi, 

I've setup an ELK environment to collect netflow records containing the MAC address in the flow record, I consistently get the following error and logstash immediately stops:

```
{:timestamp=>"2015-10-14T14:40:03.165000-0400", :message=>"Exception in inputworker", "exception"=>#<ArgumentError: wrong number of arguments calling `to_s` (1 for 0)>, "backtrace"=>["/opt/logstash
/vendor/bundle/jruby/1.9/gems/logstash-codec-netflow-1.0.0/lib/logstash/codecs/netflow/util.rb:50:in `get'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.1.0/lib/bindata/array.rb:208:in `e
ach'", "org/jruby/RubyArray.java:1613:in `each'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.1.0/lib/bindata/array.rb:208:in `each'", "org/jruby/RubyEnumerable.java:759:in `collect'", "/
opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-codec-netflow-1.0.0/lib/logstash/codecs/netflow/util.rb:50:in `get'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.1.0/lib/bindata/primit
ive.rb:111:in `sensible_default'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.1.0/lib/bindata/base_primitive.rb:142:in `_value'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.1.
0/lib/bindata/primitive.rb:103:in `do_num_bytes'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.1.0/lib/bindata/struct.rb:250:in `sum_num_bytes_below_index'", "org/jruby/RubyRange.java:479
:in `each'", "org/jruby/RubyEnumerable.java:866:in `inject'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.1.0/lib/bindata/struct.rb:247:in `sum_num_bytes_below_index'", "/opt/logstash/ven
dor/bundle/jruby/1.9/gems/bindata-2.1.0/lib/bindata/struct.rb:243:in `sum_num_bytes_for_all_fields'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.1.0/lib/bindata/struct.rb:141:in `do_num_
bytes'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.1.0/lib/bindata/base.rb:174:in `num_bytes'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-codec-netflow-1.0.0/lib/logstash/cod
ecs/netflow.rb:186:in `decode_netflow9'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-codec-netflow-1.0.0/lib/logstash/codecs/netflow.rb:86:in `decode'", "/opt/logstash/vendor/bundle/jruby
/1.9/gems/bindata-2.1.0/lib/bindata/array.rb:208:in `each'", "org/jruby/RubyArray.java:1613:in `each'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/bindata-2.1.0/lib/bindata/array.rb:208:in `each'"
, "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-codec-netflow-1.0.0/lib/logstash/codecs/netflow.rb:85:in `decode'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-udp-1.0.0/lib/l
ogstash/inputs/udp.rb:95:in `inputworker'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-udp-1.0.0/lib/logstash/inputs/udp.rb:74:in `udp_listener'"], :level=>:error}
```
This happens after the first record after receiving the template. 

Checking the offending line I can see in util.rb:

```
  class MacAddr < BinData::Primitive
    array :bytes, :type => :uint8, :initial_length => 6

    def set(val)
       ints = val.split(/:/).collect { |int| int.to_i(16) }
       self.bytes = ints
    end

    def get
      self.bytes.collect { |byte| byte.to_s(16) }.join(":")
    end
  end
```

I must start saying that I'm not expert in ruby but the message seemed to complain about the number of parameters sent to to_s so I changed

    self.bytes.collect { |byte| byte.to_s(16) }.join(":")
to 

    self.bytes.collect { |byte| byte.to_s }.join(":")

And my collection is now working and properly parsing the records.

**Comment by jorritfolmer** [comment]
You can also check PR #23

**Comment by jorritfolmer** [comment]
Closed with merge of #23
